### PR TITLE
LSIF: Use repository ids instead of names

### PR DESCRIPTION
**Pull request by efritz**
Replace usages of repository names in lsif-server with the repository ID known by the frontend. Closes https://github.com/sourcegraph/sourcegraph/issues/6278.

- *schema changes*: these changes will do a join on the `repo` table so that the repository ids are populated with the id of the repository that has the same name. If a repository id cannot be correlated (due to a rename that has already occurred), those uploads are removed.
  - rename lsif_uploads.repository to repository_name_at_upload
  - add new field lsif_uploads.repository_id
  - add new field lsif_commits.repository_id
  - drop lsif_commits.repository
  - remove all indexes/constraints from repository and add symmetric ones to repository_id (both tables)
- *lsif-server api changes*:
  - accept repository_id and (optionally) repository_name instead of repository for all query endpoints
  - accept repository_id and (required, see see [Slack thread](https://sourcegraph.slack.com/archives/CHXHX7XAS/p1579105544057100)) repository_name instead of repository for the upload endpoint
- *lsif-server internal changes*: use repository_id everywhere where repository was used before (except for gitserver interfacing, which needs to use the repository name value, if supplied to the API)
- *lsif-server tests*: this is where most of the diff is - but they're all trivial differences
- *graphql api changes*: remove InputRepoName, as we no longer have this data and recent changes to the uploads graphql api has made it an unnecessary datapoint (you must specify a repo in order to list uploads in the first place)
- *web changes*: update usage of the graphql api

### CODE REVIEW

**Comment by efritz** [comment]
> Original design decision behind lsif was to not be in the same DB as repo. You are now doing joins on it, which will make it very tricky to undo.

I made it a point that lsif-server only touches the lsif-server tables. The repo table is only joined during migrations, which are guaranteed to run on the same machine. There are no such queries at runtime (unless I missed a point somewhere). Not doing this data transition during migrations would cost a manual step with a network call for every unique repo that has LSIF data.

**Comment by efritz** [comment]
> Why do you still thread through repo name in many places with the ID?

Because when we interface with gitserver we still need to know the name (both to calculate the shard and to actually perform the request). This can be simplified in a cleanup PR after https://github.com/sourcegraph/sourcegraph/issues/7812.

**Comment by keegancsmith** [comment]
> The repo table is only joined during migrations, which are guaranteed to run on the same machine.

Aah of course. Great :D

File: web/src/enterprise/repo/settings/backend.tsx [hunk 1]
```diff
@@ -79,6 +78,37 @@ export function fetchLsifUploads({
     )
 }
 
+/**
+ * Fetch repository name by ID.
+ */
+export function fetchRepositoryName({ id }: { id: string }): Observable<string | null> {
+    return queryGraphQL(
+        gql`
+            query LsifUpload($id: ID!) {
```
Comment (chrismwendt) [inline web/src/enterprise/repo/settings/backend.tsx]:
```suggestion
            query FetchRepositoryName($id: ID!) {
```

File: web/src/enterprise/repo/settings/RepoSettingsLsifUploadPage.tsx [hunk 2]
```diff
@@ -88,7 +88,7 @@ export const RepoSettingsLsifUploadPage: FunctionComponent<Props> = ({
                                             {uploadOrError.projectRoot.commit.repository.name}
                                         </Link>
                                     ) : (
-                                        uploadOrError.inputRepoName
+                                        repositoryName
```
Comment (eseliger) [inline web/src/enterprise/repo/settings/RepoSettingsLsifUploadPage.tsx]: I think this should have a `isErrorLike(repositoryName) ? <ErrorAlert> : repositoryName` or similar

Comment (efritz) [reply web/src/enterprise/repo/settings/RepoSettingsLsifUploadPage.tsx]: Ooh good catch!

**Pull request merged**